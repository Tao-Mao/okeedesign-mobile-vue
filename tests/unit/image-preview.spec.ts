import { mount, triggerDrag, later, trigger } from '../index';
import { ImagePreview as NativeImagePreview } from '@src';
const ImagePreview = NativeImagePreview as any;
trigger;
later;
triggerDrag;

// @ts-ignore
import pic from '../pics/pic.svg';
// @ts-ignore
import pic2 from '../pics/pic2.svg';
// @ts-ignore
import pic3 from '../pics/pic3.svg';
// @ts-ignore
import pic4 from '../pics/pic4.svg';

const images = [pic, pic2, pic3, pic4];
describe('ImagePreview', () => {
  beforeEach(() => {
    // @ts-ignore
    jest.spyOn(window, 'requestAnimationFrame').mockImplementation(cb => cb());
  });
  it('render image', () => {
    const onOpen = jest.fn();
    const onChange = jest.fn();
    const wrapper = mount(ImagePreview, {
      propsData: {
        images,
        value: true,
        swipeDuration: 2,
        startIndex: 0,
      },
      listeners: {
        change: onChange,
        open: onOpen,
      },
    }) as any;
    const swipe = wrapper.find('.byted-image-preview__swipe');
    trigger(swipe, 'touchstart', 0, 0);
    trigger(swipe, 'touchmove', -100, 0);
    trigger(swipe, 'touchmove', -500, 0);
    trigger(swipe, 'touchend', -500, 0);
    expect(onChange).toBeCalled();
  });

  test('components', async () => {
    const wrapper = mount({
      template: `
        <byted-image-preview
          v-model="show"
          :swipeDuration="100"
          :images="images"
          :startIndex="1"
          @opened="onOpened"
          @change="onChange"
          @open="onOpen" />
      `,
      components: {
        'byted-image-preview': ImagePreview,
      },
      data() {
        return {
          images,
          show: false,
        };
      },
      methods: {},
    });
    wrapper.setData({
      show: true,
    });
    await later();
    expect(wrapper).toMatchSnapshot();
  });
  afterEach(() => {
    // @ts-ignore
    window.requestAnimationFrame.mockRestore();
  });
});
