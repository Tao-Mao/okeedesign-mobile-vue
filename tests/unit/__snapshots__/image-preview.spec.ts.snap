// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImagePreview components 1`] = `
<transition-stub name="byted-fade" style="z-index: 2003;">
  <div class="byted-image-preview">
    <div class="byted-image-preview__swipe">
      <div class="byted-image-preview__item"><img src="test-file-stub" class="byted-image-preview__image"></div>
      <div class="byted-image-preview__item" style="top: 0px; left: 0px;"><img src="test-file-stub" class="byted-image-preview__image"></div>
      <div class="byted-image-preview__item"><img src="test-file-stub" class="byted-image-preview__image"></div>
      <div class="byted-image-preview__item"><img src="test-file-stub" class="byted-image-preview__image"></div>
    </div>
    <div class="byted-image-preview__indicators"><i class="byted-image-preview__indicator-item"></i><i class="byted-image-preview__indicator-item byted-image-preview__indicator-item--active"></i><i class="byted-image-preview__indicator-item"></i><i class="byted-image-preview__indicator-item"></i></div>
  </div>
</transition-stub>
`;
